src/System.Net.Http/tests/FunctionalTests/DefaultCredentialsTest.cs

```diff
@@ -153,5 +198,50 @@ private class CredentialWrapper : ICredentials
             public NetworkCredential GetCredential(Uri uri, String authType) => 
                 InnerCredentials?.GetCredential(uri, authType);
         }
+
+        private class AuthenticatedProxy : IWebProxy
+        {
+            ICredentials _credentials;
+            Uri _proxyUri;
+
+            public AuthenticatedProxy(ICredentials credentials)
+            {
+                _credentials = credentials;
+
+                string host = Environment.GetEnvironmentVariable("HTTP_INTERNAL_PROXYSERVER");
```

Comment:
Nit: Maybe it would be better to keep all these variables in a common file. Otherwise there's a risk of creating other variables for the same purpose in other tests (e.g. WebSockets).